CriticalMass/MapViewController.swift

```diff
@@ -280,6 +285,64 @@ extension MapViewController: MKMapViewDelegate {
         }
         return renderer
     }
+
+    // MARK: Menu interaction handling
+
+    func mapView(_: MKMapView, didSelect view: MKAnnotationView) {
+        if view.annotation is CriticalMassAnnotation {
+            let longTapGestureRecognizer = UILongPressGestureRecognizer(
+                target: self,
+                action: #selector(handleEventLongPress(_:))
+            )
+            view.addGestureRecognizer(longTapGestureRecognizer)
+        }
+    }
+
+    @objc private func handleEventLongPress(_: Any) {
+        let routeAction = UIAlertAction(title: L10n.menuRoute, style: .default) { _ in self.routeToEvent() }
+        let shareAction = UIAlertAction(title: L10n.menuShare, style: .default) { _ in self.shareEvent() }
+        AlertPresenter.shared.presentAlert(
+            title: L10n.menuTitle,
+            preferredStyle: .actionSheet,
+            actionData: [routeAction, shareAction],
+            isCancelable: true
+        )
+    }
+
+    private func routeToEvent() {
+        nextRideManager.nextRide.flatMap {
+            $0.openInMaps()
+        }
+    }
+
+    private func shareEvent() {
+        nextRideManager.nextRide.flatMap {
+            let activityViewController = UIActivityViewController(
+                activityItems: [$0.shareMessage],
+                applicationActivities: nil
+            )
+            self.present(activityViewController, animated: true, completion: nil)
```

Comment:
same here

Reply:
?

Reply:
It's also about the flatMap here. I'm not a fan of using it for unwrapping an optional (except for when it used to transform a collection into another collection)

Reply:
But it’s totally valid code since it s not an operator for collections but rather for optionals and I think in a case where we are only interested if a value exists it is even improving readability.
What does you make not a big fan?

I can change it to`if let` but I’m not convinced yet by the argument’s